pkg/monitor/nvidia/cudevshr.go

```diff
@@ -245,11 +301,123 @@ func loadCache(fpath string) (*ContainerUsage, error) {
 	return usage, nil
 }
 
-func isValidPod(name string, pods *corev1.PodList) bool {
-	for _, val := range pods.Items {
-		if strings.Contains(name, string(val.UID)) {
+// Initialize the informer with the specified resync interval
+func (l *ContainerLister) initInformerWithConfig(resyncInterval time.Duration) error {
+	nodename := os.Getenv(util.NodeNameEnvName)
+	if nodename == "" {
+		return fmt.Errorf("env %s not set", util.NodeNameEnvName)
+	}
+
+	// Create informer factory with a longer resync period to reduce API calls
+	l.informerFactory = informers.NewSharedInformerFactoryWithOptions(
+		l.clientset,
+		resyncInterval,
+		informers.WithTweakListOptions(func(options *metav1.ListOptions) {
+			options.FieldSelector = fmt.Sprintf("spec.nodeName=%s", nodename)
+		}),
+	)
+
+	l.podInformer = l.informerFactory.Core().V1().Pods().Informer()
```

Comment:
@miaobyte I think you can create a Pod Lister here by `l.podLister := l.podInformer.Lister()`, And replace original code `pods, err := l.clientset.CoreV1().Pods("").List(...)` with `pods, err := l.podLister.Pods("").List(...)`